src/Infrastructure/Process/FileCopier/SymfonyFileCopier.php

```diff
@@ -39,34 +43,41 @@ public function copy(
             // timeout, so we have to enforce it ourselves.
             set_time_limit((int) $timeout);
 
-            $iterator = $this->createIterator($from);
+            $iterator = $this->createIterator($from, $exclusions);
             $this->filesystem->mirror($from, $to, $iterator);
         } catch (IOException $e) {
             throw new ProcessFailedException($e->getMessage(), (int) $e->getCode(), $e);
         }
     }
 
     /**
+     * @param string[] $exclusions
+     *
      * @throws \PhpTuf\ComposerStager\Exception\DirectoryNotFoundException
      */
-    private function createIterator(string $from): RecursiveCallbackFilterIterator
+    private function createIterator(string $from, array $exclusions): Finder
     {
         try {
-            $directoryIterator = new RecursiveDirectoryIterator($from);
-        } catch (UnexpectedValueException $e) {
-            throw new DirectoryNotFoundException($from, 'The "copy from" directory does not exist at "%s"', (int) $e->getCode(), $e);
-        }
-
-        /** @var callable(mixed, mixed, \RecursiveIterator<mixed, mixed>):bool $callback */
-        $callback = function (SplFileInfo $current): bool {
-            // Ignore current and parent directories.
-            if (in_array($current->getFilename(), ['.', '..'], true)) {
-                return false;
-            }
+            $this->finder
+                ->in($from)
+                ->filter(function (SplFileInfo $current) use ($from, $exclusions): bool {
+                    // Make path name relative to "from" path like exclusions.
```

Comment:
Do we need to use a callback here? Finder has a lot of powerful built-in filtering, like `notPath()` and things like that. I don't object to using the callback; I'm just wondering if we're reinventing the wheel a bit. If we're not, maybe a comment would be handy to explain what the callback is doing that precludes the use of Finder's features.

Reply:
_Ooh..._ Good call, @phenaproxima -- that simplifies things significantly!